src/legacy/core_plugins/elasticsearch/lib/health_check.js

```diff
@@ -18,23 +18,20 @@
  */
 
 import Promise from 'bluebird';
-import elasticsearch from 'elasticsearch';
 import kibanaVersion from './kibana_version';
 import { ensureEsVersion } from './ensure_es_version';
 
-const NoConnections = elasticsearch.errors.NoConnections;
-
-export default function (plugin, server) {
-  const config = server.config();
-  const callAdminAsKibanaUser = server.plugins.elasticsearch.getCluster('admin').callWithInternalUser;
-  const REQUEST_DELAY = config.get('elasticsearch.healthCheck.delay');
+export default function (plugin, server, requestDelay) {
+  const adminCluster = server.plugins.elasticsearch.getCluster('admin');
```

Comment:
**question:** as I understand we **must** support this contract for BWC. Are there any plans to introduce clear separation between `service` and `plugin` in future?

Reply:
> as I understand we **must** support this contract for BWC.

Yes, exactly because of BWC.

> Are there any plans to introduce clear separation between `service` and `plugin` in future?

We should kill legacy elasticsearch plugin as soon as possible. New platform plugins/services will use es service as a part of the contracts and legacy world may want to switch to `server.core.elasticsearch` instead as soon as we're happy with the Elasticsearch service API.